src/VirtualClient/VirtualClient.Common/Extensions/CollectionExtensions.cs

```diff
@@ -42,7 +42,14 @@ public static void AddRange<T>(this IDictionary<string, T> dictionary, IEnumerab
                 }
                 else
                 {
-                    dictionary.Add(entry.Key, entry.Value);
+                    try
```

Comment:
As I mentioned in the other comment, we need to be a little careful about this. We don't want some class to have extra parameter that doesn't apply to it. Each dictionary should be separate, and if you are somehow merging and have conflicts, the caller should decide whether to ignore this error. 

Also imagine this profile:
One workload has a Command: "-abc"
The other workload has command: undefined (user wants default)
By allowing parameters of one workload flow to others, you now allow the first parameter to overwrite the next one

Reply:
Added a Global Parameter